### PR TITLE
ImportBundles: set condition to allow importing `dsnpKeys` and `keyPairs` without schemaId

### LINKED ISSUE #168: [bug] Unable to import DSNP public keys without schema ID, and import with empty pages still clears graph state
1. As a client of the GraphSDK API, I should be able to import a user's public DSNP graph keys ONLY, without importing the entire graph. Currently the import function assumes that `schema_id` is present in the import bundle

2. If I import only public DSNP  keys with a zero-length vector of pages, the existing graph state for the user is still cleared. I should be able to import only public keys; If no pages are present in the import bundle, the graph state should be untouched. _(Note: if it is desired to be able to clear a user graph, it could be removed and re-added to the graph state, or an additional `clear_user_graph` method could be created)_

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is to set condition to allow importing `dsnpKeys` and `keyPairs` without schemaId

Closes #168 

# Checklist
- [x] Tests added

### CODE REVIEW

File: core/src/api/api.rs [hunk 1]
```diff
@@ -417,11 +417,20 @@ impl GraphState {
...
+				Some(connection_type) => connection_type,
+				None => {
+					if pages.is_empty() && !key_pairs.is_empty() && !dsnp_keys.is_none() {
+						// This condition implies that only keys are being imported
+						ConnectionType::Follow(PrivacyType::Private)
+					} else {
+						return Err(DsnpGraphError::InvalidSchemaId(*schema_id))
+					}
+				},
+			};
```
Comment (saraswatpuneet) [inline core/src/api/api.rs:432]: Here the very specific condition is checked
if we dont have schemaId and pages are empty then if and only if `dsnpKeys` and `keyPairs` are provided, we proceed else throw the error

Comment (saraswatpuneet) [reply core/src/api/api.rs:432]: In this part, its more like a safe bypass for only importing the keys

File: bridge/node/js/graph.test.ts [hunk 2]
```diff
@@ -483,4 +488,62 @@ describe("Graph tests", () => {
...
+      ownerDsnpUserId: "1000",
+      dsnpKeys,
+      connection: connectionPrivate,
+    }
+    const appliedAction = await  graph.applyActions([privateConnectAction]);
+    expect(appliedAction).toEqual(true);
+    const exported = graph.exportUpdates();
+    expect(exported).toBeDefined();
+    expect(exported.length).toEqual(1);
+  });
```
Comment (saraswatpuneet) [inline bridge/node/js/graph.test.ts:548]: This test, import bundles with empty pages, 0 schemaId and then create a private connection to ensure we are able to use those keys and export updates

File: core/src/api/api.rs [hunk 3]
```diff
@@ -417,11 +417,20 @@ impl GraphState {
...
-				.get_connection_type_from_schema_id(*schema_id)
-				.ok_or(DsnpGraphError::InvalidSchemaId(*schema_id))?;
+			let connection_type_option =
+				self.environment.get_config().get_connection_type_from_schema_id(*schema_id);
+			let connection_type = match connection_type_option {
+				Some(connection_type) => connection_type,
+				None => {
+					if pages.is_empty() && (!key_pairs.is_empty() || !dsnp_keys.is_none()) {
+						// This condition implies that only keys are being imported
+						ConnectionType::Follow(PrivacyType::Private)
```
Comment (aramikm) [inline core/src/api/api.rs:427]: I'm not sure about this, it doesn't look that clean

Comment (saraswatpuneet) [reply core/src/api/api.rs:427]: I am divided here too, however this connection_type is not used anywhere in the rest of the logic under these conditions

We can skip it away if pages are empty? Looking into some cleaner solution

Comment (saraswatpuneet) [reply core/src/api/api.rs]: Tweaked it a bit to be cleaner, let me know your thoughts

**Review by aramikm — APPROVED** [review]
Looks better. Thanks.

**Pull request merged**